File: lessons(start_android_Java)/P0101_Listener/app/src/main/java/com/example/p0101_listener/MainActivity.java
package com.example.p0101_listener;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    TextView textV;
    Button b1, b2, b3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textV = findViewById(R.id.textView);
        b1 = findViewById(R.id.button1);
        b2 = findViewById(R.id.button2);
        b3 = findViewById(R.id.button3);

        b1.setOnClickListener(listener);
        b2.setOnClickListener(listener);
        b3.setOnClickListener(listener);

        textV.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                b3.setText("you are named button4 now duddee");
            }
        });
    }

    // one Click Listener for all objects(buttons)
    // it is for sake of optimization
    View.OnClickListener listener = new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            switch (view.getId()) {
                case R.id.button1:
                    // one of ways to do this (res/values/strings.xml)
                    textV.setText(R.string.text1);
                    break;
                case R.id.button2:
                    textV.setText("222привет");
                    break;
                case R.id.button3:
                    textV.setText("333привет");
                    break;
            }
        }

    };
}

File: lessons(neco_ru_Kotlin)/lessons_1-5(OnClickListener&Button)/app/src/main/java/com/example/test_1/MainActivity.kt
package com.example.test_1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() { // need for 4th and 5th ways! (, View.OnClickListener)
    val MYLOG = "LogMainAct"

    lateinit var myTView: TextView
    lateinit var myButton: Button


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        myTView = findViewById(R.id.myTView)
        myButton = findViewById(R.id.myButton)


        // there are 5 ways of setting Listener

        /*
        // 1st way
        myButton.setOnClickListener {
            myTView.setText("hi")
        }
        */

       /*
       // 2nd way
        myButton.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View?) {
                myTView.setText("hi")
            }

        })
        */

        /*
        // 3rd way
        myButton.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View?) {
                myTView.setText("hi")
            }

        })
        */

        /*
        // 4th way (use interface)
        myButton.setOnClickListener(this)
        */

        /*
        // 5th way
        myButton.setOnClickListener(listener)
        */

        Log.d(MYLOG,"onCreate")
    }

   /*
   //4th way
    override fun onClick(view: View?) {
        when(view?.id){
            R.id.myButton->{
                myTView.setText("hi")
            }
        }
    }
    */

    /*
    // 5th way
    val listener = View.OnClickListener { view->
        when (view.getId()){
            R.id.myButton -> myTView.setText("hi")
        }
    }
    */

    override fun onStart() {
        super.onStart()
        Log.d(MYLOG,"onStart")
    }
    override fun onResume() {
        super.onResume()
        Log.d(MYLOG,"onResume")
    }

    override fun onRestart() {
        super.onRestart()
        Log.d(MYLOG,"onRestart")
    }

    override fun onPause() {
        super.onPause()
        Log.d(MYLOG,"onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.d(MYLOG,"onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(MYLOG,"onDestroy")
    }
}